### PR TITLE
Feature/init

### PR DESCRIPTION
**Pull request by qby-ill**
# Description

Initialization of the Repo
# Change overview (tick true):

- [x] This introduces backward incompatible changes
- [ ] This adds a new backward compatible Feature
- [ ] This fixes a Bug

# Version information: 

- Previous Version: `v0.0.1`

- Next Version based on [Semantic Versioning](https://semver.org/#summary) (see above): `v1.0.0`

# How Has This Been Tested?

- [x] Apply of all examples was successfull

# Checklist:

- [x] I have run tests and documented them above
- [x] I have performed a self-review of my own code
- [x] I have updated the documentation

### CODE REVIEW

File: .terraform-docs.yml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  Over this Module, named [archetype-lib](https://registry.terraform.io/modules/qbeyond/archetype-lib/azurerm/latest) all q.beyond archetype-, policy-, policy set- and roledefinitions and policy assignments will be summarized in one folder with all the azurerm compatible definitions and assignments.
+  That folder containing the external and the q.beyond libraries. The planed usecase is to hand the output `merged_library` of this Module to the CAF-Module, as it can only handle one Folder as input containing definitions and assignements.
+  
+  ## Repository structure
+
+  ## Examples
+
+  ### Basic
+  The given Template is first adding this Module as "archetype_lib". The Path of the library, that is supposed to be set as the value from `cutomer_lib`. 
+  The output "file_names" can be used to manualy check, wether all Files that are supposed to be included in the merged library are include, as it outputs the name of all files that were added.
```
Comment (TheodorWulff) [inline .terraform-docs.yml:56]: manually check whether all files ... are included ... names

File: .terraform-docs.yml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  This [repository](https://github.com/qbeyond/terraform-azurerm-archetype-lib) is our central library of custom policy (set) definitions. Policies defined here will be usable by CAF archetypes in the future.
+  Over this Module, named [archetype-lib](https://registry.terraform.io/modules/qbeyond/archetype-lib/azurerm/latest) all q.beyond archetype-, policy-, policy set- and roledefinitions and policy assignments will be summarized in one folder with all the azurerm compatible definitions and assignments.
+  That folder containing the external and the q.beyond libraries. The planed usecase is to hand the output `merged_library` of this Module to the CAF-Module, as it can only handle one Folder as input containing definitions and assignements.
+  
+  ## Repository structure
+
+  ## Examples
+
+  ### Basic
+  The given Template is first adding this Module as "archetype_lib". The Path of the library, that is supposed to be set as the value from `cutomer_lib`.
```
[the remark this answers is not in the document]
Comment (TheodorWulff) [reply .terraform-docs.yml:55]: I am a bit confused by the second sentence, I think its missing a verb

File: .terraform-docs.yml [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  That folder containing the external and the q.beyond libraries. The planed usecase is to hand the output `merged_library` of this Module to the CAF-Module, as it can only handle one Folder as input containing definitions and assignements.
+  
+  ## Repository structure
+
+  ## Examples
+
+  ### Basic
+  The given Template is first adding this Module as "archetype_lib". The Path of the library, that is supposed to be set as the value from `cutomer_lib`. 
+  The output "file_names" can be used to manualy check, wether all Files that are supposed to be included in the merged library are include, as it outputs the name of all files that were added.
+  The output of `merged_library` is suppoed to be the Input for the [CAF-Module](https://registry.terraform.io/modules/aztfmod/caf/azurerm/latest) for the variable named `archetype_lib`
```
Comment (TheodorWulff) [inline .terraform-docs.yml:57]: supposed to ... input .. for the parameter `archetype_lib`.

File: main.tf [hunk 4]
```diff
@@ -0,0 +1,22 @@
+locals {
+  customer_lib         = trimsuffix(var.customer_lib, "/")
+  definiton_files      = fileset(local.customer_lib, "**/{archetype_definition,policy_assignment,policy_definition,policy_set_definition,role_definition}_*.{json,yml,yaml,json.tftpl,yml.tftpl,yaml.tftpl}")
```
Comment (TheodorWulff) [inline main.tf:3]: Rename to `definition_files`

File: main.tf [hunk 5]
```diff
@@ -0,0 +1,22 @@
+locals {
+  customer_lib         = trimsuffix(var.customer_lib, "/")
+  definiton_files      = fileset(local.customer_lib, "**/{archetype_definition,policy_assignment,policy_definition,policy_set_definition,role_definition}_*.{json,yml,yaml,json.tftpl,yml.tftpl,yaml.tftpl}")
+  definiton_files_path = toset([for path in local.definiton_files : "${local.customer_lib}/${path}"])
```
Comment (TheodorWulff) [inline main.tf:4]: Same here: `definition_files_path`

File: main.tf [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+resource "local_file" "copied_files" {
+  for_each = local.definiton_files_path
+  source   = each.value
+  filename = "${path.module}/customer_lib/${basename(each.key)}"
+}
+
+output "merged_library" {
+  value       = path.module
+  description = "Path to where the library, containing both libraries can be found. This output can be given to the CAF-Module."
```
Comment (TheodorWulff) [inline main.tf:15]: No comma needed

File: main.tf [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  depends_on  = [local_file.copied_files]
+}
+
+output "file_names" {
+  value       = local.definiton_files
+  description = "Outputs the files wich were added to the library."
```
Comment (TheodorWulff) [inline main.tf:21]: which

File: variables.tf [hunk 8]
```diff
@@ -0,0 +1,4 @@
+variable "customer_lib" {
+  type        = string
+  description = "Path to the customer libaray folder containing definition files that are supposed to be used by the CAF-Module. This module picks all CAF compatible definitions from the given folder and its subdirectories."
```
Comment (TheodorWulff) [inline variables.tf:3]: library

File: .terraform-docs.yml [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+      {{- template "setDict" dict "Dict" $resourceTypes "Key" (printf "%s_%s" .ProviderName .Type) "Resource" . -}}
+    {{- end -}}
+  {{- end -}}
+  {{ indent 0 "#" }}
+  
+  # Archetype Library
+
+  ## Introduction
+  This [repository](https://github.com/qbeyond/terraform-azurerm-archetype-lib) is our central library of custom policy (set) definitions. Policies defined here will be usable by CAF archetypes in the future.
+  Over this Module, named [archetype-lib](https://registry.terraform.io/modules/qbeyond/archetype-lib/azurerm/latest) all q.beyond archetype-, policy-, policy set- and roledefinitions and policy assignments will be summarized in one folder with all the azurerm compatible definitions and assignments.
```
[the remark this answers is not in the document]
Comment (qby-ill) [reply .terraform-docs.yml:47]: done

File: .terraform-docs.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (qby-ill) [reply .terraform-docs.yml:55]: changed to:
 The Path of the library that is supposed to be summorized with the q.beyond library, is set as the value of `cutomer_lib`.

File: .terraform-docs.yml (hunk 1 above)
Comment (qby-ill) [reply .terraform-docs.yml:56]: done :D

File: .terraform-docs.yml (hunk 3 above)
Comment (qby-ill) [reply .terraform-docs.yml:57]: done :)

File: main.tf (hunk 4 above)
Comment (qby-ill) [reply main.tf:3]: done :p

File: main.tf (hunk 5 above)
Comment (qby-ill) [reply main.tf:4]: same here, done :p

File: main.tf (hunk 6 above)
Comment (qby-ill) [reply main.tf:15]: now the comma is sad, but it is gone :(_

File: main.tf (hunk 7 above)
Comment (qby-ill) [reply main.tf:21]: done

File: variables.tf (hunk 8 above)
Comment (qby-ill) [reply variables.tf:3]: also done, thank you :D

File: .terraform-docs.yml [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+    {{- if $isResource -}}
+      {{- template "setDict" dict "Dict" $resourceTypes "Key" (printf "%s_%s" .ProviderName .Type) "Resource" . -}}
+    {{- end -}}
+  {{- end -}}
+  {{ indent 0 "#" }}
+  
+  # Archetype Library
+
+  ## Introduction
+  This [repository](https://github.com/qbeyond/terraform-azurerm-archetype-lib) is our central library of custom policy (set) definitions. Policies defined here will be usable by CAF archetypes in the future.
```
Comment (QBY-MarkusMaring) [inline .terraform-docs.yml:46]: ~~future~~

File: .terraform-docs.yml [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+  Over this Module, named [archetype-lib](https://registry.terraform.io/modules/qbeyond/archetype-lib/azurerm/latest), all q.beyond archetype-, policy-, policy set- and roledefinitions and policy assignments will be summarized in one folder with all the azurerm compatible definitions and assignments.
+  That folder containing the external and the q.beyond libraries. The planed usecase is to hand the output `merged_library` of this Module to the CAF-Module, as it can only handle one Folder as input containing definitions and assignements.
```
Comment (QBY-MarkusMaring) [inline .terraform-docs.yml:48]: will contain, ~~libraries~~ definitions

File: .terraform-docs.yml [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+      - they each contain their needed policy-, policy set- and roledefinitions and policy assignments
+  - .tf files that are the archetype-lib modulue
+  - folder for examples
+    - contains folders of examples on how to use the module. Each example is described in this README under "Examples"
+
+
+  ## Examples
+
+  ### Basic
+  First, the given template is adding this Module as `archetype_lib`. The Path of the library that is supposed to be summorized with the q.beyond library, is set as the value of `cutomer_lib`.
```
Comment (QBY-MarkusMaring) [inline .terraform-docs.yml:64]: summarized

File: .gitignore [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+
+# Include override files you do wish to add to version control using negated pattern
+# !example_override.tf
+
+# Include tfplan files to ignore the plan output of command: terraform plan -out=tfplan
+# example: *tfplan*
+
+# Ignore CLI configuration files
+.terraformrc
+terraform.rc
```
[the remark this answers is not in the document]
Comment (qby-ill) [reply .gitignore:34]: muss noch eingebaut werden, aber ich ignoriere die Datei schonmal

File: .terraform-docs.yml [hunk 14]
```diff
@@ -0,0 +1,119 @@
...
+  
+  # Archetype Library
+
+  ## Introduction
+  This [repository](https://github.com/qbeyond/terraform-azurerm-archetype-lib) is our central library of custom policy (set) definitions. Policies defined here will be usable by CAF archetypes in the future.
+  Over this Module, named [archetype-lib](https://registry.terraform.io/modules/qbeyond/archetype-lib/azurerm/latest), all q.beyond archetype-, policy-, policy set- and roledefinitions and policy assignments will be summarized in one folder with all the azurerm compatible definitions and assignments.
+  That folder containing the external and the q.beyond libraries. The planed usecase is to hand the output `merged_library` of this Module to the CAF-Module, as it can only handle one Folder as input containing definitions and assignements.
+  
+  ## Root Repository Structure
+  - archetypes is a folder just containgn archetypedefinitions: no subfolders, the name should be describing enough
```
[the remark this answers is not in the document]
Comment (qby-ill) [reply .terraform-docs.yml:51]: changed to 
"no subfolders, the name serves as description"
is that fine?

**Review by QBY-MarkusMaring — DISMISSED** [review]

File: README.md [hunk 15]
```diff
@@ -1,2 +1,84 @@
-# azure-archetype-lib
-Repository with q.beyond designed Azure-archetypes, including policies
+# Archetype Library
+[![GitHub tag](https://img.shields.io/github/tag/qbeyond/terraform-azurerm-archetype-lib.svg)](https://registry.terraform.io/modules/qbeyond/terraform-azurerm-archetype-lib/provider/latest)
+[![License](https://img.shields.io/github/license/qbeyond/terraform-azurerm-archetype-lib.svg)](https://github.com/qbeyond/terraform-azurerm-archetype-lib/blob/main/LICENSE)
+
+----
+
+This [repository](https://github.com/qbeyond/terraform-azurerm-archetype-lib) is our central library of custom policy (set) definitions. Policies defined here will be usable by CAF archetypes.
+Over this Module, named [archetype-lib](https://registry.terraform.io/modules/qbeyond/archetype-lib/azurerm/latest), all q.beyond archetype-, policy-, policy set- and roledefinitions and policy assignments will be summarized in one folder with all the azurerm compatible definitions and assignments.
```
Comment (TheodorWulff) [inline README.md:8]: Via this module

File: README.md [hunk 16]
```diff
@@ -1,2 +1,84 @@
...
+That folder will containin the external and the q.beyond definitions and assignments. The planed usecase is to hand the output `merged_library` of this Module to the CAF-Module, as it can only handle one Folder as input containing definitions and assignements.
```
Comment (TheodorWulff) [inline README.md:9]: planned*

Comment (TheodorWulff) [reply README.md:9]: assignments*

File: README.md [hunk 17]
```diff
@@ -1,2 +1,84 @@
...
+
+## Root Repository Structure
+- archetypes is a folder just containing archetype definitions: no subfolders, the name serves as description
+- The main categories used by archetypes: policy_definition, role_definitions
+  - they each contain folders named by the topics that the policys or roles are for
```
Comment (TheodorWulff) [inline README.md:14]: policies*

Comment (TheodorWulff) [reply README.md:14]: Very german phrase, try: ... named by the corresponding topics of the policies and roles*

File: README.md [hunk 18]
```diff
@@ -1,2 +1,84 @@
...
+    - they each can contain a folder with tests for themselves
+    - they each contain their needed policy-, policy set- and roledefinitions and policy assignments
+- .tf files that are the archetype-lib module
```
Comment (TheodorWulff) [inline README.md:17]: that make up the

File: README.md [hunk 19]
```diff
@@ -1,2 +1,84 @@
...
+
+<!-- BEGIN_TF_DOCS -->
+##
+
+# Archetype Library
+
+## Examples
+
+### Basic
+First, the given template is adding this Module as `archetype_lib`. The Path of the library that is supposed to be summarized with the q.beyond library, is set as the value of `cutomer_lib`.
```
Comment (TheodorWulff) [inline README.md:29]: No comma

**Review by TheodorWulff — CHANGES_REQUESTED** [review]

**Pull request merged**